waku/protocol/v2/waku_protocol.nim

```diff
@@ -26,7 +26,7 @@ type
   WakuSub* = ref object of GossipSub
     # XXX: just playing
     text*: string
-    gossip_enabled*: bool
+    pubsub*: Pubsub
```

Comment:
Seems fine to me, does it definitely work though? Have you tried it with both floodsub and gossipsub and simulator etc behaves correctly?

Reply:
Looking at the procCalls I don't see any reason why it wouldn't, but that also seems to assume function calls/arguments stay the same. I suppose it can still be worked around though.

@kdeme how does this toggling between flood/gossipsub work out in NBC?

Reply:
afaik this switch is part of `newStandardSwitch` and just set fixed to gossip.